docs/modules/ROOT/pages/headers.adoc

```diff
@@ -0,0 +1,54 @@
+# DataSonnet Header
+
+The header is an optional comment section in the mapping document which must start with `/\** DataSonnet` followed by a new line character and end with a line `*/` and contains a set of configuration properties.
+The properties are hierarchical, with sections separated by the `.` (dot) character, e.g.:
+
+------------
+dataformat.application/xml.payload.NamespaceSeparator=:
+dataformat.application/xml.payload.TextValueKey=__text
+dataformat.application/csv.myVar.Quote="
+dataformat.application/csv.myVar.Separator=|
+------------
+
+The header must be the first comment before any other line of code, in the following format:
+
+-------------
+/** DataSonnet
+version=<VERSION_NUMBER>
```

Comment:
We should specify how DataSonnet will handle header versions in detail. Probably worth discussing in the issue.

Probably the version should be the code version? Or rather, the "language version", which isn't quite the same thing, but will roughly track code version. Probably every time language version is ticked it is immediately caught up to the code version, which will tick more often. Things that would tick language version: changes in built in libraries and syntax (including header syntax). Things that would not: details of the plugins, as different ones can be loaded.

And then we should behavior under different misalignments of the header version and the runtime version. Which likely are mostly matters of what warnings to print when the header version is older, and might include not running the code at all when the header version is much newer?